cmd/shared-lock.go

```diff
@@ -40,6 +40,7 @@ func (ld sharedLock) backgroundRoutine(ctx context.Context, objAPI ObjectLayer,
 			continue
 		}
 
+	keepLock:
```

Comment:
Shouldn't we break out to the beginning of the outer for loop, so that we can reacquire the shared lock? We could still get into the same hot loop, since `<-lkctx.Context().Done()` won't block unless the shared lock is reacquired.

Reply:
@vadmeste ^^

Reply:
> Shouldn't we break out to the beginning of the outer for loop, so that we can reacquire the shared lock?

I think this PR is doing exactly that @krisis

Reply:
You're right @vadmeste. I had incorrectly assumed that `break` statement worked similar to `goto`.